components/dependency_injection/advanced.rst

```diff
@@ -218,3 +218,56 @@ You can change the inner service name if you want to:
             ->addArgument(new Reference('bar.wooz'))
             ->setPublic(false)
             ->setDecoratedService('foo', 'bar.wooz');
+
+Deprecating Services
+--------------------
+
+Once you have decided to deprecate the use of a service (because it is outdated,
+or you decided not to use and maintain it anymore), you can deprecate its
+definition:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+       bar:
+         class: stdClass
+         deprecated: true
+         deprecation_template: The "%service_id%" service is deprecated since 2.8, and will be removed in 3.0.
+
+    .. code-block:: xml
+
+        <service id="bar" class="stdClass">
+            <deprecated>The "%service_id%" service is deprecated since 2.8, and will be removed in 3.0.</deprecated>
+        </service>
+
+    .. code-block:: php
+
+        use Symfony\Component\DependencyInjection\Reference;
+
+        $container->register('bar', 'stdClass')
+            ->setDeprecated(true, 'The "%service_id%" service is deprecated since 2.8, and will be removed in 3.0.');
+
+Now, every time a service is created using this deprecated definition will
+trigger a deprecation error, advising you to stop or change your uses of that
```

Comment:
[...] definition, a deprecation warning will be triggered [...] or to change [...]